src/FocusApp.Client/Methods/User/EditUserProfile.cs

```diff
@@ -0,0 +1,66 @@
+﻿using FocusApp.Client.Clients;
+using FocusApp.Client.Helpers;
+using FocusApp.Shared.Data;
+using FocusCore.Commands.User;
+using FocusCore.Responses;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace FocusApp.Methods.User;
+public class EditUserProfile
+{
+    internal class Handler : IRequestHandler<EditUserProfileCommand, MediatrResult>
+    {
+        FocusAppContext _localContext;
+        ILogger<Handler> _logger;
+        IAuthenticationService _authenticationService;
+        IAPIClient _client;
+        public Handler(FocusAppContext localContext, ILogger<Handler> logger, IAuthenticationService authenticationService, IAPIClient client)
+        {
+            _localContext = localContext;
+            _logger = logger;
+            _authenticationService = authenticationService;
+            _client = client;
+        }
+
+        public async Task<MediatrResult> Handle(EditUserProfileCommand command, CancellationToken cancellationToken)
+        {
+            try
+            {
+                // Update the user on the server
+                await _client.EditUserProfile(command, default);
```

Comment:
Since a cancellation token was passed could we use it here instead of the default?